azure-iot-hub/tests/test_iothub_registry_manager.py

```diff
@@ -0,0 +1,800 @@
+        hostname=fake_hostname,
+        device_id=fake_device_id,
+        skn=fake_shared_access_key_name,
+        sk=fake_shared_access_key,
+    )
+    iothub_registry_manager = IoTHubRegistryManager(connection_string)
+    return iothub_registry_manager
+
+
+@pytest.mark.describe("IoTHubRegistryManager")
+class TestIoTHubRegistryManagerInit(object):
+    @pytest.mark.it("instantiates correctly with auth")
+    def test_init_auth(self, iothub_registry_manager):
+        assert iothub_registry_manager is not None
+        assert iothub_registry_manager.auth is not None
+        assert isinstance(iothub_registry_manager.auth, ConnectionStringAuthentication)
+
+    @pytest.mark.it("instantiates correctly with protocol client with correct Service API version")
+    def test_init_protocol_client(self, iothub_registry_manager):
+        assert iothub_registry_manager is not None
+        assert iothub_registry_manager.protocol is not None
+        assert isinstance(iothub_registry_manager.protocol, IotHubGatewayServiceAPIs20190630)
+
+
+different_devices_sas = [
+    pytest.param(
+        {
+            "function_name": "create_device_with_sas",
+            "init_kwargs": {
+                "device_id": fake_device_id,
+                "primary_key": fake_primary_key,
+                "secondary_key": None,
+                "status": fake_status,
+            },
+            "etag": None,
+        },
+        id="Create with Primary Key",
+    ),
+    pytest.param(
+        {
+            "function_name": "create_device_with_sas",
+            "init_kwargs": {
+                "device_id": fake_device_id,
+                "primary_key": None,
+                "secondary_key": fake_secondary_key,
+                "status": fake_status,
+            },
+            "etag": None,
+        },
+        id="Create with Secondary Key",
+    ),
+    pytest.param(
+        {
+            "function_name": "update_device_with_sas",
+            "init_kwargs": {
+                "device_id": fake_device_id,
+                "primary_key": fake_primary_key,
+                "secondary_key": None,
+                "status": fake_status,
+                "etag": fake_etag,
+            },
+            "etag": fake_etag,
+        },
+        id="Update with Primary Key",
+    ),
+    pytest.param(
+        {
+            "function_name": "update_device_with_sas",
+            "init_kwargs": {
+                "device_id": fake_device_id,
+                "primary_key": None,
+                "secondary_key": fake_secondary_key,
+                "status": fake_status,
+                "etag": fake_etag,
+            },
+            "etag": fake_etag,
+        },
+        id="Update with Secondary Key",
+    ),
+]
+
+
+@pytest.mark.parametrize("params_methods_sas", different_devices_sas)
+@pytest.mark.describe("IoTHubRegistryManager Create/Update Device With Symmetric Key")
```

Comment:
>"IoTHubRegistryManager Create/Update Device With Symmetric Key" [](start = 22, length = 63)

Please follow the naming and unit structure conventions for tests as outlined in the wiki:

[https://github.com/Azure/azure-iot-sdk-python/wiki/unittesting_unit_examples](https://github.com/Azure/azure-iot-sdk-python/wiki/unittesting_unit_examples)